### ISSUE TITLE
Buttons cannot be customized: Can we make them generic over a containing widget?

### ISSUE THREAD

**Issue by twitchyliquid64** [linebender/druid#2159]
As it stands, the current implementation of `widget::Button` cannot be meaningfully customized. Some things that would be nice:

 - Containing other visual cues, such as a icon
 - Increasing the size for buttons that are meant to be prominent

Proposal:

The current implementation nests a `Label<T>` but aside from `Button::dynamic()` (which we can keep as its a builder method) doesn't do anything label-specific in layout/paint. What if we made `Button` generic over its contained widget?

That would allow us to keep the button-specific logic we know and love (the hovering, rounded edges, click visuals etc) but enable custom contents as well as some control over sizing.

Happy to send a PR if this works.

**Comment by twitchyliquid64** [comment]
I agree, it would be nice to have a clearer strat on what the next-gen improvements & patterns are across our widget base. Agree that discussion might make more sense as a supplementary issue. Do you have something in mind?

At least for this issue around buttons, I'm fairly confident that this fits the existing model of composing widgets within other widgets through a builder method, delegating the non-core events/logic down to descendant widgets while handling whats important for itself (in a buttons case, thats the hover effect, the rounded corners, and the `on_click` handler).

**Comment by nickdbush** [comment]
I've popped the [comment above](https://github.com/linebender/druid/issues/2159#issuecomment-1081007534) into a new issue (#2160) – if you have any thoughts regarding the above, do add them there as well!
